test/sqllogictest/arrays.slt

```diff
@@ -852,3 +852,64 @@ SELECT ARRAY[ARRAY[customer.first_name], ARRAY[customer.zip], ARRAY[customer.id:
 query error db error: ERROR: array_in not yet supported
 SELECT array_in('', 0, 0)
 ----
+
+# Array casting for column expressions
```

Comment:
Can you add some tests of the following format:

- Us `pg_tyepof` to assert that the final type is correct.
- Error scenarios where casts fail.

Reply:
Definitely, added.